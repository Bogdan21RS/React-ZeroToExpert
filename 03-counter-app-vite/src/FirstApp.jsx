export default function FirstApp() {
  return (
    <>
      <h1>First App</h1>
      <p>This is a paragraph</p>
    </>
  );
}
