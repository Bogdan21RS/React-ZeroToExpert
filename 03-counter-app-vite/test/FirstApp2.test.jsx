import { render, screen } from "@testing-library/react";
import FirstApp from "../src/firstApp";

describe("Pruebas de FirstApp", () => {
  const title = "Hola Mundo";
  test("Should match with the snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show the message 'Hello World' ", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Should show the title in h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });
});
