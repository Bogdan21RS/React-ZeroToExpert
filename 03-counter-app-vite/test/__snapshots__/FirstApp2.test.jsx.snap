// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de FirstApp Should match with the snapshot 1`] = `
<div>
  <h1>
    Hola Mundo
  </h1>
  <h2>
    subtitulo
  </h2>
  <p>
    This is a paragraph
  </p>
</div>
`;
